### DISCUSSION TITLE
Modal issue with orphaned content between slots

### DISCUSSION THREAD

**Discussion by Otisssss** [livewire/livewire#4017] [7. Bugs (archived)]
Love the package.. thanks for the hardwork. Not sure if this is a bug in livewire, my component but figured somewhere would be great to make note. I just spent a couple hours debugging this.  

Below is the troubled spot in a datatable with form modal..

`           <form wire:submit.prevent="deleteItemSelected">`
        `<x-modal.confirmdelete wire:model.defer="showDeleteModal">`
            `<x-slot name="title">Delete Item</x-slot>asfsdfsdfsf`
           ` <x-slot name="content">`

Any content between the two slots like asfsdfsdfsf will cause the modal (in this case) to always popup on load. Might be something to take a look at I was surprised at this myself.

**Comment by joshhanley** [comment]
@Otisssss please share all the relevant code needed to replicate the issue, and if you can provide some kind of demonstration of it, using something like https://gifcap.dev/

Thanks!

**Reply by Otisssss** [reply]
@joshhanley attached is a gifcap. I start by showing it working normally. Then you see me add the text and then refresh. It pops the modal. It completely doesn't respect any validation rules, etc..
![recording](https://user-images.githubusercontent.com/87203931/137390042-f38dd8e1-4da3-483e-beb3-c49cc358dee9.gif)

**Reply by joshhanley** [reply]
@Otisssss thanks! Any chance you could create a small example repo which reproduces the issue with as little code as possible?

**Reply by Otisssss** [reply]
@joshhanley I was actually testing this based off of the datatables from the surge repo which I got access for being a subscriber.  If you have access to that you can quickly test it out. It would be in the I believe dashboard blade.

**Comment by joshhanley** [comment]
@Otisssss finally got a chance to have a look into this, and it turns out that it is a Laravel blade compiler issue.

I stripped down the surge dashboard to the bare minimum and managed to work out what is going on.

Basically what happens is if you have orphaned content at the end of the closing `x-slot` tag, the blade compiler breaks.

So for example this:
```blade
<x-slot name="some-slot">Some content</x-slot>SOMETEXT
```

Doesn't actually render the end slot properly, and looks like this

![image](https://user-images.githubusercontent.com/882837/142503088-5f3afab2-09fc-4b6a-8e8e-acb2301197e1.png)

To fix this, if you just put a space between the closing slot tag and the text, then it all functions correctly.

```blade
</x-slot> SOMETEXT
```

![image](https://user-images.githubusercontent.com/882837/142503244-7e230fd2-f94b-4bd3-a663-0083aca298b3.png)

So bascially when Laravel's blade compiler converts `</x-slot>` into the raw `@endslot` directive, there is no space after it, causing it not to close the slot, hence triggering other rendering issues (like the modal options not being rendered properly, hence why it was displayed by default).

As it is though, that's something that would need to be fixed in Laravel (if at all) 🙂

Hope this helps!
